### PR TITLE
feat: add support for new volume types

### PR DESCRIPTION
**Pull request by backjo**
Signed-off-by: Jonah Back <jonah@jonahback.com>

### CODE REVIEW

File: api/v1alpha1/instancegroup_types.go [hunk 1]
```diff
@@ -111,6 +111,12 @@ var (
 	LifecycleHookAllowedDefaultResult   = []string{LifecycleHookResultAbandon, LifecycleHookResultContinue}
 	LaunchTemplatePlacementTenancyTypes = []string{HostPlacementTenancyType, DefaultPlacementTenancyType, DedicatedPlacementTenancyType}
 	log                                 = ctrl.Log.WithName("v1alpha1")
+
+	volumeTypesWithProvisionedIopsSupport = map[string]bool{
```
Comment (eytan-avisror) [inline api/v1alpha1/instancegroup_types.go:115]: minor nit:
Should we move this to awsprovider along with `AllowedVolumeTypes`?
We might want a shorter name too - how about `AllowedVolumeTypesWithProvisionedIOPS`?

Also what are you thoughts on using a string slice (such as `AllowedVolumeTypes`) vs. a map like you implemented here?

We can then remove the map check and change the if condition to something like:
```yaml
if v.Iops != 0 && !common.ContainsEqualFold(awsprovider.AllowedVolumeTypesWithProvisionedIOPS, v.Type) {
  ...
}
```

This is minor and both options are good, but just might be more in line with existing code.

**Comment by eytan-avisror** [comment]
LG, added some minor suggestion to refactor.
Assume you verified manually this works? 🥇

File: api/v1alpha1/instancegroup_types.go (hunk 1 above)
Comment (backjo) [reply api/v1alpha1/instancegroup_types.go:115]: Sounds good - will make this check!

**Comment by backjo** [comment]
Yep, verified this works on our EKS clusters. One caveat that might be applicable for some folks is that the LaunchConfiguration APIs don't support the new volume types - only the LaunchTemplate APIs

**Comment by eytan-avisror** [comment]
> Yep, verified this works on our EKS clusters. One caveat that might be applicable for some folks is that the LaunchConfiguration APIs don't support the new volume types - only the LaunchTemplate APIs

In that case, should we add another validation above the provisioned IOPS one to validate if user is trying to use invalid types with template/config?

We can add `TemplateAllowedVolumeTypes` & `ConfigurationAllowedVolumesTypes`
You can use predicates `spec.IsLaunchTemplate()` or `spec.IsLaunchConfiguration()` to check what the CR is declaring.

**Comment by backjo** [comment]
Added validation based on ScalingConfigurationType

**Review by eytan-avisror — APPROVED** [review]
LG
Thanks!

**Pull request merged**